Add accessibility info to <progress> doc (#10707)
mdn/content: files/en-us/web/html/element/progress/index.md

On Windows 7, the resulting progress looks like this:

![progress-firefox.JPG](progress-firefox.jpg)
  
## Accessibility Concerns
  
### Labelling
  

> In most cases you should provide an accessible label when using `<progress>`. While you can use the standard ARIA labelling attributes [`aria-labelledby`](https://www.w3.org/TR/wai-aria/#aria-labelledby) or [`aria-label`](https://www.w3.org/TR/wai-aria/#aria-label) as you would for any element with `role="progressbar"`, when using `<progress>` you can alternatively use the {{htmlelement("label")}} element.

Author: I wasn't sure if I should refer to the ARIA attributes using this syntax or one of the HTML attribute macros (https://github.com/mdn/yari/blob/main/kumascript/macros/htmlattrxref.ejs, https://github.com/mdn/yari/blob/main/kumascript/macros/htmlattrdef.ejs).

Went with this approach because it looks like MDN doesn't currently have pages those macros would link to for `aria-label` and `aria-labelledby` (see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes), although there are "Using the x attribute" pages for both (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute, https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute)

Please advise if this is fine or if a different approach is preferred.

Reviewer: The [aria-label page](https://github.com/mdn/content/pull/10575) is currently in PR phase, so should be live next week. Aria-labelledby is up next.
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes shows the pages we have live on sidebar nav (below the content on narrow screens). The main content area has dead links to the pages that will be written in the next few weeks. There is no macro .... yet.

Author: So should we just link to those non-existent pages for now since they'll be published soon rather than the spec?

Reviewer: yes please 😄

Author: resolved in 1ac5b49

  
> **Note:** Text placed between the element's tags is not an accessible label, it is only recommended as a fallback for old browsers that do not support this element.

#### Example
  
```html
  <label>Uploading Document: <progress value="70" max="100">70 %</progress></label>
  
  <!-- OR -->
  
  <label for="progress-bar">Uploading Document</label>
  <progress id="progress-bar" value="70" max="100">
```
    
### Describing a particular region

> If the `<progress>` element is describing the loading progress of a particular region of a page, you **SHOULD** use [`aria-describedby`](https://www.w3.org/TR/wai-aria-1.1/#aria-describedby) to point to the status, and set the [`aria-busy`](https://www.w3.org/TR/wai-aria-1.1/#aria-busy) attribute to true on the region until it is finished loading.

Author: This is copied almost verbatim from [the aria spec](https://www.w3.org/TR/wai-aria-1.2/#progressbar) and from the MDN doc on [using the progressbar role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role). I spent a bit trying to link to the latter for more, but ended up deciding to just copy the text here as pretty much the rest of that document is really only applicable if not using `<progress>`. Happy to change the approach if somebody has a better idea.

Reviewer: ```suggestion
If the `<progress>` element is describing the loading progress of a section of a page, use [`aria-describedby`](en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) to point to the status, and set [`aria-busy="true"`](en-US/docs/Web/Accessibility/ARIA/Attributes/aria_busy ) on the section that is being updated, removing the `aria-busy` attribute when it has finished loading.
```

Reviewer: all the roles and attributes with pages in the aria_techniques section are being rewritten with new pages in en-US/docs/Web/Accessibility/ARIA/Attributes  and en-US/docs/Web/Accessibility/ARIA/Roles. The spec is a little bit difficult to read, which is part of the reason MDN exists. I tried to simplify the language just a bit. I didn't reword the word "status" because I am not 100% what it refers to. I think this would benefit from that word being explained.

Author: > I didn't reword the word "status" because I am not 100% what it refers to. I think this would benefit from that word being explained.

The way I understood it was that "status" referred to the `progress` element.

Author: After reading through the spec again I'm less confident that I interpreted it correctly. I could see it going either way "putting the `describedby` on the section which is loading pointing to the indicator or vice versa. Also interesting that the spec doesn't mention removing the `describedby` when the loading is done the way it mentions updating `aria-busy`. And I can't find literally ANY examples of this online, other than a few sites that have just copied the spec text verbatim.

Rather than take a guess and risk misleading people maybe @jnurthen, @joanmarie or @mcking65 could weigh in and clarify what the proper interpretation is? They seem to have been involved in semi-recent edits to that part of the spec over in https://github.com/w3c/aria.

Author: Per [this issue](https://github.com/w3c/aria/issues/1654#issuecomment-982887387), it looks like the initial interpretation in this PR was correct.
